src/components/Common/ServiceRequestTimeline.tsx

```diff
@@ -0,0 +1,133 @@
+import { CheckIcon } from "@radix-ui/react-icons";
+import React from "react";
+
+export interface ProgressBarSubStep {
+  label: string;
+  status?:
+    | "completed"
+    | "active"
+    | "pending"
+    | "collected"
+    | "received"
+    | "dispatched"
+    | "processed"
+    | "reviewed"
+    | "notStarted";
```

Comment:
this seems very specific to lab orders, lets make it generic, may be like `completed` | `in-progress` | `yet-to-start` or consider using number like percentage for progress,

here is a thought
```ts
type Step = {
  index: number;
  title: string;
  progress: number;
  subSteps: Step[];
};
```